ground-server/app/grpc/page.js

```diff
@@ -0,0 +1,27 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { runSendCommand } from './commandClient';
+
+export default function GrpcPage() {
+  const [message, setMessage] = useState('Loading...');
+
+useEffect(() => {
+    runSendCommand().then(res => {
+        try {
+            if(res.ack) {
+                setMessage("Command Acknowledged!")
+            }
+        } catch (err) {
+            setMessage("No Acknowledgement Received")
+        }
```

Comment:
The `then` function on a promise only gets called if runSendCommand is successful, so I don't think the catch block will be called. If you do something like:

```javascript
runSendCommand().then(res => {
    setMessage("Command Acknowledged!")
}).catch(err => {
    setMessage("No Acknowledgement Received")
})
```

That should catch the error properly. You can also keep the `res.ack` if statement in if you want.

Reply:
Thanks for catching this!